pkg/cluster/spec.go

```diff
@@ -204,18 +204,42 @@ func NewSpecFromClusterConfig(clusterConfigPath string, cliVersion version.Info,
 		return nil, err
 	}
 
-	kubeDistro, err := buildKubeDistro(eksd)
-	if err != nil {
+	if err = s.init(clusterConfig, bundlesManifest, versionsBundle, eksd); err != nil {
 		return nil, err
 	}
 
-	s.Bundles = bundlesManifest
-	s.Config = clusterConfig
+	switch s.Cluster.Spec.DatacenterRef.Kind {
+	case eksav1alpha1.TinkerbellDatacenterKind:
+		templateConfigs, err := eksav1alpha1.GetTinkerbellTemplateConfig(clusterConfigPath)
+		if err != nil {
+			return nil, err
+		}
+		s.TinkerbellTemplateConfigs = templateConfigs
+	}
+
+	if s.ManagementCluster != nil {
+		s.Cluster.SetManagedBy(s.ManagementCluster.Name)
+	} else {
+		s.Cluster.SetSelfManaged()
+	}
+
+	return s, nil
+}
+
+// init does the basic initialization with the provided necessary api objects
+func (s *Spec) init(config *Config, bundles *v1alpha1.Bundles, versionsBundle *v1alpha1.VersionsBundle, eksdRelease *eksdv1alpha1.Release) error {
```

Comment:
moved this piece to a separate method so it can be reused by `NewSpecFromClusterConfig` and `BuildSpec`